src/tests/multihost/ad/test_dyndns.py

```diff
@@ -0,0 +1,330 @@
+    old_hostname = cmd.stdout_text.rstrip()
+    ad_domain = session_multihost.ad[0].domainname
+    new_hostname = "client" + str(random.randrange(99))
+    new_hostname = new_hostname + "." + ad_domain
+    session_multihost.client[0].run_command(f'hostname {new_hostname}', raiseonerr=False)
+
+    def restore_hostname():
+        """ Restore hostname """
+        session_multihost.client[0].run_command(f'hostname {old_hostname}', raiseonerr=False)
+    request.addfinalizer(restore_hostname)
+
+
+@pytest.fixture(scope="function")
+def extra_network(session_multihost, request):
+    """ Create reverse zone """
+    network = "192.168.1.1"
+    network_list = network.split(".")
+    network_zone = str(network_list[2] + '.' + network_list[1] + '.' + network_list[0])
+    cmd_zoneadd = "dnscmd.exe /zoneadd %s.in-addr.arpa /primary" % network_zone
+    cmd_zoneconfig = "dnscmd.exe /config %s.in-addr.arpa /allowupdate 1" % network_zone
+    session_multihost.ad[0].run_command(cmd_zoneadd)
+    session_multihost.ad[0].run_command(cmd_zoneconfig)
+
+    def remove_extra_network():
+        """ Delete reverse zone"""
+        session_multihost.ad[0].run_command(
+            "dnscmd.exe /zonedelete %s.in-addr.arpa /f" % network_zone)
+
+    request.addfinalizer(remove_extra_network)
+
+
+@pytest.fixture(scope="function")
+def extra_interface(session_multihost, request):
+    """ Create extra interface """
+    dummy_interface = 'ibm' + str(random.randrange(99))
+    dummy_ip = '192.168.1.' + str(random.randrange(2, 255))
+    session_multihost.client[0].run_command(
+        'ip link add ' + dummy_interface + ' type dummy', raiseonerr=False)
+    session_multihost.client[0].run_command(
+        'ip addr add ' + dummy_ip + ' dev ' + dummy_interface, raiseonerr=False)
+
+    def remove_interface():
+        """ Delete interface """
+        session_multihost.client[0].run_command(
+            'ip addr flush dev ' + dummy_interface, raiseonerr=False)
+        session_multihost.client[0].run_command(
+            'ip link del ' + dummy_interface, raiseonerr=False)
+
+    request.addfinalizer(remove_interface)
+    return dummy_interface, dummy_ip
+
+
+@pytest.fixture(scope="function")
+def getptr(session_multihost, request):
+    """ Get network and ptr record """
+    ip = str(session_multihost.client[0].ip)
+    net_list = ip.split(".")
+    network = str(net_list[2] + '.' + net_list[1] + '.' + net_list[0])
+    ptr = str(net_list[3])
+
+    def removeptr():
+        cmd_delete_ptr = "dnscmd.exe /recorddelete %s.in-addr.arpa %s. PTR /f" % (network, ptr)
+        session_multihost.ad[0].run_command(cmd_delete_ptr, raiseonerr=False)
+    request.addfinalizer(removeptr)
+
+    return ip, network, ptr
+
+
+@pytest.mark.usefixtures("change_client_hostname")
+@pytest.mark.dyndns
+@pytest.mark.tier1_3
+class TestDynDns(object):
+
+    @pytest.mark.tier1_3
+    def test_0001_verify_with_default_setting(self, multihost, adjoin, getptr):
+        """
```

Comment:
Please make sure that first line of the test docstring is a description and there is a empty line before the other fields as beetlejuice needs.